lib/inspec/plugins/resource.rb

```diff
@@ -57,12 +67,33 @@ def self.example(example = nil)
           @example = example
         end
 
+        def skip_resource(message)
+          @resource_skipped = message
+        end
+
+        def resource_skipped?
+          !@resource_skipped.nil?
+        end
+
+        def resource_skipped_message
```

Comment:
Why do we need to distinguish between failed and skip message. Can we have both at the same time?

Reply:
Hmmm...you have a point. We shouldn't have both. Should we merge both into `resource_skipped_message`? Something else? Technically a failed resource isn't skipped since it tries to execute, but I don't want to rename it to `resource_message`. Thoughts?

Reply:
Part of the approach here was to ensure that we weren't botching something that was considered to be "public" or something RSpec was expecting. I'm +1 for consolidating as long as it doesn't break any backward compatibility.

I think calling it `exception_message` is probably sufficient.

Reply:
I think merging into `exception_message` is a fine idea. Before, it was all contained in `resource_skipped` which should still work going forward.

Reply:
Well fiddlesticks...
https://github.com/chef/inspec/blob/master/test/unit/resources/nginx_conf_test.rb#L100

This tests calls the instance variable `@resource_skipped` expecting a message. Since this test does it, is it safe to assume that any user could call the instance variable directly instead of the deprecated `resource_skipped` method?

If so it's no big, I'll just keep `@resource_skipped` and `@resource_failed` set to messages and evaluate in `resource_exception_message`...but I'd really like those to be bools.

Reply:
If you call an instance variable directly, you deserve breakage. 🙂  In no way can we support that as long-term behavior.

I'd fix the test and proceed with consolidation.

Reply:
Agreed! Fixed.